sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/FilterPushdownBenchmark.scala

```diff
@@ -362,6 +394,13 @@ object FilterPushdownBenchmark extends BenchmarkBase with SQLHelper {
     }
 
     runBenchmark(s"Pushdown benchmark with many filters") {
+      // This benchmark and the next one are similar in that they both test predicate pushdown
+      // where the filter itself is very large. There have been cases where the filter conversion
+      // would take minutes to hours for large filters due to it being implemented with exponential
+      // complexity in the height of the filter tree.
+      // The difference between these two benchmarks is that this one benchmarks pushdown with a
+      // large string filter (`a AND b AND c ...`), whereas the next one benchmarks pushdown with
+      // a large Column-based filter (`col(a) || (col(b) || (col(c)...))`).
```

Comment:
If I read this comment correctly, it seems that we should just remove the next benchmark, as string filter and Column-based filter have no difference regarding performance. Is there any other critical difference that I missed?

Reply:
@cloud-fan the two go through different code paths. The string-based one was added in https://github.com/apache/spark/pull/22313 , but it doesn't expose the slowness when passing a `Column` filter directly. That is, the string-based one was fast before this PR. The one this PR fixes is specifically when passing in a `Column` directly to something like `df.filter(Column)`.

Reply:
I still can't get it. Both the string filter and column-based filter will become an `Expression` in the `Filter` operator. The differences I see are
1. the new benchmark builds a larger filter
2. the new benchmark use `Or` instead of `And`.